src/core/server/plugins/plugin_context.ts

```diff
@@ -65,6 +79,23 @@ export function createPluginInitializerContext(
      * Core configuration functionality, enables fetching a subset of the config.
      */
     config: {
+      /**
+       * Global configuration
+       * Note: naming not final here, it will be renamed in a near future (https://github.com/elastic/kibana/issues/46240)
+       * @deprecated
+       */
+      globalConfig__deprecated$: combineLatest(
+        coreContext.configService.atPath<KibanaConfigType>(kibanaConfig.path),
+        coreContext.configService.atPath<ElasticsearchConfigType>(elasticsearchConfig.path),
+        coreContext.configService.atPath<PathConfigType>(pathConfig.path)
+      ).pipe(
+        map(([kibana, elasticsearch, path]) => ({
+          kibana: pick(kibana, SharedGlobalConfigKeys.kibana),
+          elasticsearch: pick(elasticsearch, SharedGlobalConfigKeys.elasticsearch),
+          path: pick(path, SharedGlobalConfigKeys.path),
```

Comment:
I wonder if we should change this with some sort of overwrite where, instead of providing the root of the data path (ie: `fromRoot('data')`), we return a subfolder with the name of the plugin ((ie: `path.join(fromRoot('data'), pluginManifest.id)`)) so a plugin cannot access others (or at least not directly).

They could still access other plugins' data if they move up to the parent folder, but, at least 1 faulty plugin cannot directly delete all other plugins' data that easily.

NB: If we go for this subfolder option, should we automatically create the folder for the plugin, or is their responsibility to ensure it?

Reply:
I think legacy plugins currently don't really use a `myplugin` subfolder in data. This looks like breaking changes, specifically when this legacy root config accessor is supposed to be only to ease migration.

Reply:
I don't think we will provide it to the NP plugins at all. As I can see only core and optimizer consume it. Optimizer won't exist in the NP, I suppose